src/components/UI/modals/SubmitProposalModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Modal, Space, Typography } from "antd";
+import Field from "components/DefaultField/DefaultField";
+import SendButton from "components/UI/buttons/SendButton/SendButton";
+import submitProposalSchema from "components/UI/modals/SubmitProposalModelSchema";
+import { Form } from "components/UI/modals/SubmitProposalStyles";
+import { useEffect } from "react";
+import { useForm, SubmitHandler, FieldValues } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { toast } from "react-toastify";
+import { useSendProposalMutation } from "services/jobPosts/proposalsAPI";
+
+const { Text } = Typography;
+interface ISubmitProposalForm extends FieldValues {
+  bid: number;
+  exampleRequired: string;
+}
+
+interface SubmitProposalModalProps {
+  idJobPost: number;
+  visible: boolean;
+  closeModal: () => void;
+  clientBudget: number;
+}
+
+function SubmitProposalModal({
+  idJobPost,
+  visible,
+  closeModal,
+  clientBudget,
+}: SubmitProposalModalProps) {
+  const { t } = useTranslation();
+
+  const { control, handleSubmit, reset } = useForm<ISubmitProposalForm>({
+    resolver: yupResolver(submitProposalSchema),
+  });
+
+  const [runSendProposalMutation, { isError, isLoading, isSuccess }] =
+    useSendProposalMutation();
+
+  const onSubmit: SubmitHandler<ISubmitProposalForm> = async ({
+    bid,
+    coverLetter,
+  }) => {
+    await runSendProposalMutation({ bid, coverLetter, idJobPost });
+    closeModal();
```

Comment:
just suggestion. As long as u using async logic antd library provide to modal `confirmLoading` prop, so u can pass `isLoading` value, and it automaticly will disable submit button, because it looks like user can send many proposal before modal wil be closed

Reply:
I don't know if it was necessary since we have the closeModal() and reset() functions running afterwards, but I added it just in case

Reply:
but closeModal() and reset() runs after await, so they will wait promise result, and during this time user can click on button again and again